tests/common/plugins/sanity_check/recover.py

```diff
@@ -57,16 +65,18 @@ def adaptive_recover(dut, localhost, fanouthosts, check_results, wait_time):
         if result['failed']:
             logging.info("Restoring {}".format(result))
             if result['check_item'] == 'interfaces':
-                __recover_interfaces(dut, fanouthosts, result, wait_time)
-            elif result['check_item'] in ['services', 'processes']:
-                action             = __recover_services(dut, result)
-                # Only allow outstanding_action be overridden when it is
-                # None. In case the outstanding_action has already been
-                # been set to 'reboot'.
-                if not outstanding_action:
-                    outstanding_action = action
+                action = __recover_interfaces(dut, fanouthosts, result, wait_time)
+            elif result['check_item'] == 'services':
+                action = __recover_services(dut, result)
+            elif result['check_item'] == 'processes':
+                action = 'config_reload'
             else:
-                outstanding_action = 'reboot'
+                action = 'reboot'
+
+            # Any action can override no action or 'config_reload'.
+            # 'reboot' is last resort and cannot be overridden.
```

Comment:
Maybe I'm missing something, but where is this enforced? If the `outstanding_action` is `load_minigraph` for example, won't this be skipped?

Reply:
Currently, the adaptive recover didn't return load minigraph. But I think I can come back with another PR to add a action priority level in the list to only allow overriding lower priority actions.